src/util/misc.js

```diff
@@ -681,6 +687,61 @@
       };
     },
 
+    calcRotateMatrix: function(options) {
+      if (!options.angle) {
+        return fabric.iMatrix.concat();
+      }
+      var theta = fabric.util.degreesToRadians(options.angle),
+          cos = fabric.util.cos(theta),
+          sin = fabric.util.sin(theta);
+      return [cos, sin, -sin, cos, 0, 0];
+    },
+
+    calcDimensionsMatrix: function(options) {
+      var scaleX = typeof options.scaleX === 'undefined' ? 1 : options.scaleX,
+          scaleY = typeof options.scaleY === 'undefined' ? 1 : options.scaleY,
+          scaleMatrix = [scaleX, 0, 0, scaleY, 0, 0];
+      if (options.skewX) {
+        scaleMatrix = fabric.util.multiplyTransformMatrices(
+          scaleMatrix,
+          [1, 0, Math.tan(fabric.util.degreesToRadians(options.skewX)), 1],
+          true);
+      }
+      if (options.skewY) {
+        scaleMatrix = fabric.util.multiplyTransformMatrices(
+          scaleMatrix,
+          [1, Math.tan(fabric.util.degreesToRadians(options.skewY)), 0, 1],
```

Comment:
Just a little consistency question here.  Your code style uses fabric.util.cos, fabric.util.sin, but then you abruptly switch to Math.tan.  Maybe you should just define fabric.util.tan as well.

Reply:
Math.tan is good enough to be used as it is.
cos and sin never return 0 or 1 when they are at multiples of 90 degrees. That makes number go crazy, like 0.0000009999 instead of 0. That is why we have util.cos and util.sin, to return 0 and 1 when needed.